internal/command/providers_lock.go

```diff
@@ -282,6 +286,39 @@ func (c *ProvidersLockCommand) Run(args []string) int {
 			// platforms here, because the SetProvider method we call below
 			// handles that automatically.
 			hashes = append(hashes, platformLock.AllHashes()...)
+
+			// At this point, we've merged all the hashes for this (provider, platform)
+			// combo into the combined hashes for this provider. Let's take this
+			// opportunity to print out a summary for this particular combination.
+
+			if oldLock == nil {
+				// Then we weren't tracking this provider previously so we can write a simple error message.
+				madeAnyChange = true
+				c.Ui.Output(
+					fmt.Sprintf(
+						"- Obtained %s checksums for %s; This was a new provider and the checksums for this platform are now tracked in the lock file",
+						provider.ForDisplay(),
+						platform))
+				continue
+			}
+
+			if oldLock.ContainsAll(platformLock) {
```

Comment:
My understanding of this call and the function it's calling is that it will return `true` only if `platformLock` _doesn't_ add any new hashes compared to `oldLock`, which feels opposite to the message it's printing. Am I misunderstanding the behavior of `ContainsAll`? :thinking: